dkc/core/tests/test_authorized_uploads.py

```diff
@@ -0,0 +1,179 @@
+    resp = api_client.post(
+        '/api/v2/files',
+        data={
+            'name': 'foo.txt',
+            'size': 123,
+            'folder': authorized_upload.data['folder'],
+            'authorization': authorized_upload.data['signature'],
+        },
+    )
+    assert resp.status_code == 201
+    assert resp.data['creator'] == authorized_upload.data['creator']
+
+
+@pytest.mark.django_db
+def test_authorized_upload_delete_permission(api_client, user, authorized_upload):
+    api_client.force_authenticate(user)
+    resp = api_client.delete(f'/api/v2/authorized_uploads/{authorized_upload.data["id"]}')
+    assert resp.status_code == 403
+    assert resp.json() == {'detail': 'Only the creator of an authorized upload may delete it.'}
+
+
+@pytest.mark.django_db
+def test_authorized_upload_delete_as_superuser(api_client, authorized_upload, user_factory):
+    admin = user_factory(is_superuser=True)
+    api_client.force_authenticate(admin)
+    resp = api_client.delete(f'/api/v2/authorized_uploads/{authorized_upload.data["id"]}')
+    assert resp.status_code == 204
+
+
+@pytest.mark.django_db
+def test_authorized_upload_delete_as_creator(user, api_client, folder):
+    authorized_upload = AuthorizedUpload.objects.create(
+        creator=user,
+        folder=folder,
+        expires=timezone.now() + timedelta(days=1),
+    )
+    api_client.force_authenticate(user)
+    resp = api_client.delete(f'/api/v2/authorized_uploads/{authorized_upload.id}')
+    assert resp.status_code == 204
+
+
+@pytest.mark.django_db
+def test_authorized_upload_completion_bad_signature(api_client, authorized_upload):
+    resp = api_client.post(
+        f'/api/v2/authorized_uploads/{authorized_upload.data["id"]}/completion',
+        data={'authorization': f'1{authorized_upload.data["signature"]}'},
+    )
+    assert resp.status_code == 403
+    assert resp.json() == {'detail': 'Invalid authorization signature.'}
+
+
+@pytest.mark.django_db
+def test_authorized_upload_completion_id_mismatch(api_client, authorized_upload, user, folder):
+    other_upload = AuthorizedUpload.objects.create(
+        creator=user,
+        folder=folder,
+        expires=timezone.now() + timedelta(days=1),
+    )
+    resp = api_client.post(
+        f'/api/v2/authorized_uploads/{other_upload.id}/completion',
+        data={'authorization': authorized_upload.data['signature']},
+    )
+    assert resp.status_code == 403
+    assert resp.json() == {'detail': 'Invalid authorization signature.'}
+
+
+@pytest.mark.django_db
+def test_authorized_upload_completion(api_client, authorized_upload, mailoutbox):
+    resp = api_client.post(
+        f'/api/v2/authorized_uploads/{authorized_upload.data["id"]}/completion',
+        data={'authorization': authorized_upload.data['signature']},
+    )
+    assert resp.status_code == 204
+    with pytest.raises(AuthorizedUpload.DoesNotExist):
+        AuthorizedUpload.objects.get(pk=authorized_upload.data['id'])
```

Comment:
```suggestion
    assert not AuthorizedUpload.objects.filter(pk=authorized_upload.data['id']).exists()
```

Seems more straightforward to what we're asserting.